File: Examples/DotNET/CSharp/Bookmark/UpdateBookmarkText.cs
using System;
using Com.Aspose.Words.Api;
using Com.Aspose.Words.Model;
using Com.Aspose.Storage.Api;

namespace Bookmark
{
    class UpdateBookmarkText
    {
        public static void Run()
        {
            // ExStart:1
            WordsApi wordsApi = new WordsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "SampleWordDocument.docx";

            String storage = null;
            String folder = null;


            try
            {
                // Upload the file
                storageApi.PutCreate(fileName, "", storage, System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));
                // Invoke Aspose.Words Cloud SDK API to get bookmarks from document by name
                BookmarkResponse apiResponse = wordsApi.PostUpdateDocumentBookmark(fileName, "test", "", storage, folder, null);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {

                    Console.WriteLine("Bookmark updated.");

                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Node.js/Bookmark/UpdateBookmarkText.js
var StorageApi = require('asposestoragecloud');
const { WordsApi, PostDocumentSaveAsRequest, SaveOptionsData,PostUpdateDocumentBookmarkRequest,BookmarkData } = require("asposewordscloud");


var config = {'appSid':'78946fb4-3bd4-4d3e-b309-f9e2ff9ac6f9','apiKey':'b125f13bf6b76ed81ee990142d841195' , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose Words API SDK
var wordsApi = new WordsApi(config);

// Set input file name
var name = 'SampleWordDocument.docx';
	
const request = new PostUpdateDocumentBookmarkRequest();
request.name = name;
request.folder = null;
request.bookmarkName = "aspose";
request.bookmarkData = new BookmarkData({
         text: "new text"
});

wordsApi.postUpdateDocumentBookmark(request).then((result) => {
		console.log(result);  
       	}).catch(function(err) {
		console.log(err);
});
	


		
